---
title: Selected BBOP publications
---

**Please note: this page is hand-curated and may be out of date. You can find our Paperpile folder of BBOP publications at [https://paperpile.com/shared/EXYLSk](https://paperpile.com/shared/EXYLSk).**
<ol class="bibliography">
 
 <li><b>Unni DR, Moxon SAT</b>, Bada M, Brush M, Bruskiewich R, <b>Caufield JH</b> Clemons PA, Dancik V, Dumontier M, Fecho K, Glusman G, Hadlock JJ, <b>Harris NL</b>, Joshi A, Putman T, Qin G, Ramsey SA, Shefchek KA, Solbrig H, Soman K, Thessen AE, Haendel MA, Bizon C, <b>Mungall CJ</b>, The Biomedical Data Translator Consortium. Biolink Model: A universal schema for knowledge graphs in clinical, biomedical, and translational science. Clin Transl Sci. Wiley; 2022 Jun 6; https://onlinelibrary.wiley.com/doi/10.1111/cts.13302
</li><li>
Fecho K, Thessen AT, Baranzini SE, Bizon C, Hadlock JJ, Huang S, Roper RT, Southall N, Ta C, Watkins PB, Williams MD, Xu H, Byrd W, Dančík V, Duby MP, Dumontier M, Glusman G, <b>Harris NL</b>, Hinderer EW, Hyde G, Johs A, Su A, Qin G, Zhu Q, Biomedical Data Translator Consortium. Progress Toward a Universal Biomedical Data Translator. Clin Transl Sci. Wiley Online Library; 2022 May 25; http://dx.doi.org/10.1111/cts.13301 PMID: 35611543
 </li>
 <li>Danis D, Jacobsen JOB, Balachandran P, Zhu Q, Yilmaz F, <b>Reese J</b>, Haimel M, Lyon GJ, Helbig I, <b>Mungall CJ</b>, Beck CR, Lee C, Smedley D, Robinson PN. SvAnna: efficient and accurate pathogenicity prediction of coding and regulatory structural variants in long-read genome sequencing. Genome Med. 2022 Apr 28;14(1):44. http://dx.doi.org/10.1186/s13073-022-01046-6 PMID: 35484572
  </li><li>Jacobsen JO, Kelly C, Cipriani V, Consortium GER, <b>Mungall CJ, Reese J</b>, Danis D, Robinson PN, Smedley D. Phenotype-driven approaches to enhance variant prioritization and diagnosis of rare disease. Hum Mutat. Wiley; 2022 Apr 7; https://onlinelibrary.wiley.com/doi/10.1002/humu.2438 PMID: 35391505
 </li><li>Matentzoglu N, Balhoff JP, Bello SM, Bizon C, Brush M, Callahan TJ, Chute CG, Duncan WD, Evelo CT, Gabriel D, Graybeal J, Gray A, Gyori BM, Haendel M, Harmse H, <b>Harris NL</b>, Harrow I, <b>Hegde H</b>, Hoyt AL, Hoyt CT, Jiao D, Jiménez-Ruiz E, Jupp S, Kim H, Koehler S, Liener T, Long Q, Malone J, McLaughlin JA, McMurry JA, <b>Moxon S</b>, Munoz-Torres MC, Osumi-Sutherland D, Overton JA, Peters B, Putman T, Queralt-Rosinach N, Shefchek K, Solbrig H, Thessen A, Tudorache T, Vasilevsky N, Wagner AH, <b>Mungall CJ</b>. A Simple Standard for Sharing Ontological Mappings (SSSOM). Database. 2022 May 25;2022. http://dx.doi.org/10.1093/database/baac035 PMID: 35616100.
  </li><li>Sternberg PW, Agapite J, …, <b>Mungall CJ</b>, …, Zhou P, Zytkovicz M, The Alliance of Genome Resources Consortium. Harmonizing model organism data in the Alliance of Genome Resources. Genetics. Oxford University Press (OUP); 2022 Feb 25; https://academic.oup.com/genetics/advance-article/doi/10.1093/genetics/iyac022/6537086
 </li><li>Hu, B., Canon, S., Eloe-Fadrosh, E. A., Anubhav, Babinski, M., Corilo, Y., Davenport, K., Duncan, W. D., Fagnan, K., Flynn, M., Foster, B., Hays, D., …, Mouncey, N., <b>Mungall, C.J.</b>, … Chain, P. S. G. (2022). Challenges in Bioinformatics Workflows for Processing Microbiome Omics Data at Scale. Frontiers in Bioinformatics, 1. https://doi.org/10.3389/fbinf.2021.826370
 </li><li>Nadendla S, Jackson R, Munro J, Quaglia F, Mészáros B, Olley D, Hobbs ET, Goralski SM, Chibucos M, <b>Mungall CJ</b>, Tosatto SCE, Erill I, Giglio MG. ECO: the Evidence and Conclusion Ontology, an update for 2022. Nucleic Acids Res. Oxford University Press; 2021 Nov 19; https://academic.oup.com/nar/advance-article-abstract/doi/10.1093/nar/gkab1025/6431816
 </li><li>Eloe-Fadrosh, E.A., Ahmed, F., Anubhav, Babinski, M., Baumes, J., Borkum, M., Bramer, L., Canon, S., Christianson, D.S., Corilo, Y.E. Davenport, K.W., Davis, B., Drake, M., Duncan, W.D., … Chain, P.S.G., McCue, L.A., Mans, D., <b>Mungall, C.J.</b>, Mouncey, N.J., Fagnan, K.T. (2022). The National Microbiome Data Collaborative Data Portal: an integrated multi-omics microbiome data resource, Nucleic Acids Research, Volume 50, Issue D1, 7 January 2022, Pages D828–D836, https://doi.org/10.1093/nar/gkab990 
  </li><li>Coleman B, Casiraghi E, Blau H, Chan L, Haendel MA, Laraway B, Callahan TJ, Deer RR, Wilkins KJ, <b>Reese J</b>, Robinson PN. Risk of new-onset psychiatric sequelae of COVID-19 in the early and late post-acute phase. World Psychiatry. Wiley; 2022 Jun;21(2):319–320. https://onlinelibrary.wiley.com/doi/10.1002/wps.20992 PMID: 35524622

</li><li>Ravanmehr V, Blau H, Cappelletti L, Fontana T, Carmody L, Coleman B, George J, <b>Reese J, Joachimiak M</b>, Bocci G, Hansen P, Bult C, Rueter J, Casiraghi E, Valentini G, <b>Mungall C</b>, Oprea TI, Robinson PN. Supervised learning with word embeddings derived from PubMed captures latent knowledge about protein kinases and cancer. NAR Genom Bioinform. 2021 Dec;3(4):lqab113. http://dx.doi.org/10.1093/nargab/lqab113 PMCID: PMC8652379
  
  </li><li>Good BM, Van Auken K, Hill DP, Mi H, <b>Carbon S</b>, Balhoff JP, Albou L-P, Thomas PD, <b>Mungall CJ</b>, Blake JA, D’Eustachio P. Reactome and the Gene Ontology: Digital convergence of data resources. Bioinformatics. 2021 May 8; http://dx.doi.org/10.1093/bioinformatics/btab325 PMID: 33964129
    
 </li><li><b>Reese J</b>, Coleman B, Chan L, Callahan TJ, Cappelletti L, Fontana T, Bradwell KR, <b>Harris NL</b>, Casiraghi E, Valentini G, Karlebach G, Deer R, McMurry JA, Haendel MA, Chute CG, Pfaff E, Moffitt R, Spratt H, Singh J, <b>Mungall CJ</b>, Williams AE, Robinson PN. Cyclooxygenase inhibitor use is associated with increased COVID-19 severity. medRxiv, 2021. http://dx.doi.org/10.1101/2021.04.13.21255438

</li><li>Vangay P, Burgin J, Johnston A, Beck KL, Berrios DC, Blumberg K, Canon S, Chain P, Chandonia J-M, Christianson D, Costes SV, Damerow J, <b>Duncan WD</b>, Dundore-Arias JP, Fagnan K, Galazka JM, Gibbons SM, Hays D, Hervey J, Hu B, Hurwitz BL, Jaiswal P, <b>Joachimiak MP</b>, Kinkel L, Ladau J, Martin SL, McCue LA, Miller K, Mouncey N, <b>Mungall </b>C, Pafilis E, Reddy TBK, Richardson L, Roux S, Shaffer JP, Sundaramurthi JC, Thompson LR, Timme RE, Zheng J, Wood-Charlson EM, Eloe-Fadrosh EA. Microbiome Metadata Standards: Report of the National Microbiome Data Collaborative’s Workshop and Follow-On Activities. mSystems. American Society for Microbiology Journals; 2021 Feb 23;6(1). https://msystems.asm.org/content/6/1/e01194-20

</li><li>Köhler S, Gargano M, Matentzoglu N, Carmody LC, Lewis-Smith D, Vasilevsky NA, Danis D, ..., Yüksel Z, Helbig I, <b>Mungall CJ</b>, Haendel MA, Robinson PN. The Human Phenotype Ontology in 2021. Nucleic Acids Res. 2021 Jan 8;49(D1):D1207–D1217. http://dx.doi.org/10.1093/nar/gkaa1043 PMCID: PMC7778952

</li><li><b>Joachimiak MP</b>. Zinc against COVID-19? Symptom surveillance and deficiency risk groups. PLoS Negl Trop Dis. 2021 Jan;15(1):e0008895. http://dx.doi.org/10.1371/journal.pntd.0008895 PMCID: PMC7781367

</li><li>Gene Ontology Consortium. The Gene Ontology resource: enriching a GOld mine. Nucleic Acids Res. 2020 Dec 8; http://dx.doi.org/10.1093/nar/gkaa1113 PMID: 33290552

</li><li><b>Reese JT</b>, Unni D, Callahan TJ, Cappelletti L, Ravanmehr V, <b>Carbon S</b>, Shefchek KA, Good BM, Balhoff JP, Fontana T, Blau H, Matentzoglu N, <b>Harris NL</b>, Munoz-Torres MC, Haendel MA, Robinson PN, <b>Joachimiak MP, Mungall CJ</b>. KG-COVID-19: a framework to produce customized knowledge graphs for COVID-19 response. Patterns (N Y). 2020 Nov 9;100155. http://dx.doi.org/10.1016/j.patter.2020.100155 PMCID: PMC7649624

</li><li>Thessen AE, Walls RL, Vogt L, Singer J, Warren R, Buttigieg PL, Balhoff JP, <b>Mungall CJ</b>, McGuinness DL, Stucky BJ, Yoder MJ, Haendel MA. Transforming the study of organisms: Phenomic data models and knowledge bases. PLoS Comput Biol. 2020 Nov;16(11):e1008376. http://dx.doi.org/10.1371/journal.pcbi.1008376 PMID: 33232313

</li><li>Rubinstein YR, Robinson PN, Gahl WA, Avillach P, Baynam G, Cederroth H, Goodwin RM, Groft SC, Hansson MG, <b>Harris NL</b>, Huser V, Mascalzoni D, McMurry JA, Might M, Nellaker C, Mons B, Paltoo DN, Pevsner J, Posada M, Rockett-Frase AP, Roos M, Rubinstein TB, Taruscio D, van Enckevort E, Haendel MA. The case for open science: rare diseases. Jamia Open. Oxford University Press; 2020 Sep 11; https://academic.oup.com/jamiaopen/advance-article-abstract/doi/10.1093/jamiaopen/ooaa030/5904414

</li><li>Robinson PN, Ravanmehr V, Jacobsen JOB, Danis D, Zhang XA, Carmody LC, Gargano MA, Thaxton CL, UNC Biocuration Core, Karlebach G, <b>Reese J</b>, Holtgrewe M, Köhler S, McMurry JA, Haendel MA, Smedley D. Interpretable Clinical Genomics with a Likelihood Ratio Paradigm. Am J Hum Genet. 2020 Sep 3;107(3):403–417. http://dx.doi.org/10.1016/j.ajhg.2020.06.021 PMCID: PMC7477017

</li><li>Wood V, <b>Carbon S</b>, Harris MA, Lock A, Engel SR, Hill DP, Van Auken K, Attrill H, Feuermann M, Gaudet P, Lovering RC, Poux S, Rutherford KM, <b>Mungall CJ</b>. Term Matrix: a novel Gene Ontology annotation quality control system based on ontology term co-annotation patterns. Open Biol. 2020 Sep;10(9):200149. http://dx.doi.org/10.1098/rsob.200149 PMCID: PMC7536087

</li><li>Wood-Charlson EM, Anubhav, Auberry D, Blanco H, Borkum MI, Corilo YE, Davenport KW, Deshpande S, Devarakonda R, Drake M, <b>Duncan WD</b>, Flynn MC, Hays D, Hu B, Huntemann M, Li P-E, Lipton M, Lo C-C, Millard D, Miller K, Piehowski PD, Purvine S, Reddy TBK, Shakya M, Sundaramurthi JC, Vangay P, Wei Y, Wilson BE, Canon S, Chain PSG, Fagnan K, Martin S, McCue LA, <b>Mungall CJ</b>, Mouncey NJ, Maxon ME, Eloe-Fadrosh EA. The National Microbiome Data Collaborative: enabling microbiome science. Nat Rev Microbiol. nature.com; 2020 Jun;18(6):313–314. http://dx.doi.org/10.1038/s41579-020-0377-0 PMID: 32350400

</li><li>Alliance of Genome Resources Consortium. Alliance of Genome Resources Portal: unified model organism research platform. Nucleic Acids Res. 2020 Jan 8;48(D1):D650–D658. http://dx.doi.org/10.1093/nar/gkz813 PMCID: PMC6943066

</li><li>Hoyt C, <b>Mungall </b>C, Vasilevsky N, Domingo-Fernández D, Healy M, Colluru V. Extension of Roles in the ChEBI Ontology. ChemRxiv; 2020;


</li><li>Shefchek KA, <b>Harris NL</b> Gargano M, Matentzoglu N, Unni D, Brush M, Keith D, Conlin T, Vasilevsky N, Zhang XA, Balhoff JP, Babb L, Bello SM, Blau H, Bradford Y, <b>Carbon S</b> Carmody L, Chan LE, Cipriani V, Cuzick A, Rocca MD, <b>Dunn N</b>, Essaid S, Fey P, Grove C, Gourdine J-P, Hamosh A, Harris M, Helbig I, Hoatlin M, <b>Joachimiak M</b> Jupp S, Lett KB, Lewis SE, McNamara C, Pendlington ZM, Pilgrim C, Putman T, Ravanmehr V, <b>Reese J</b> Riggs E, Robb S, Roncaglia P, Seager J, Segerdell E, Similuk M, Storm AL, Thaxon C, Thessen A, Jacobsen JOB, McMurry JA, Groza T, Köhler S, Smedley D, Robinson PN, <b>Mungall CJ</b> Haendel MA, Munoz-Torres MC, Osumi-Sutherland D. The Monarch Initiative in 2019: an integrative data and analytic platform connecting phenotypes to genotypes across species. Nucleic Acids Res, 2019 Nov 8; http://dx.doi.org/10.1093/nar/gkz997 PMID: 31701156

</li><li>Haendel M, Vasilevsky N, Unni D, Bologa C, <b>Harris N</b> Rehm H, Hamosh A, Baynam G, Groza T, McMurry J, Dawkins H, Rath A, Thaxon C, Bocci G, <b>Joachimiak MP</b> Köhler S, Robinson PN, <b>Mungall C</b> Oprea TI. How many rare diseases are there? Nat Rev Drug Discov, 2019 Nov 5. https://www.nature.com/articles/d41573-019-00180-y

</li><li>Mabee PM, Balhoff JP, Dahdul WM, Lapp H, <b>Mungall CJ</b> Vision TJ. A Logical Model of Homology for Comparative Biology. Syst. Biol. 2019, p. 588822. http://dx.doi.org/10.1093/sysbio/syz067

</li><li>Thomas PD, Hill DP, Mi H, Osumi-Sutherland D, Van Auken K, <b>Carbon S</b> Balhoff JP, Albou L-P, Good B, Gaudet P, Lewis SE, <b>Mungall CJ</b>. Gene Ontology Causal Activity Modeling (GO-CAM) moves beyond GO annotations to structured descriptions of biological functions and systems. Nat Genet, nature.com; 2019 Oct;51(10):1429–1433. http://dx.doi.org/10.1038/s41588-019-0500-1 PMID: 31548717

</li><li>The Alliance of Genome Resources Consortium. The Alliance of Genome Resources: Building a Modern Data Ecosystem for Model Organism Databases. Genetics. 2019. p. 1189–1196. http://dx.doi.org/10.1534/genetics.119.302523
</li><li>Jackson RC, Balhoff JP, <b>Douglass E, Harris NL, Mungall CJ</b>, Overton JA. ROBOT: A Tool for Automating Ontology Workflows. BMC Bioinformatics , 2019 Jul 29;20(1):407. http://dx.doi.org/10.1186/s12859-019-3002-3 PMID: 31357927
</li><li><b>Carbon S</b> Champieux R, McMurry JA, Winfree L, Wyatt LR, Haendel MA. An analysis and metric of reusable data licensing practices for biomedical resources. PLoS One. 2019 Mar 27;14(3):e0213090. http://dx.doi.org/10.1371/journal.pone.0213090 PMID: 30917137
</li><li><b>Dunn NA, Unni D</b>, Diesh C, Munoz-Torres M, <b>Harris NL</b>, Yao E, Rasche H, Holmes IH, Elsik CG, Lewis SE. Apollo: Democratizing genome annotation. PLoS Comput Biol. 2019 Feb;15(2):e1006790. http://dx.doi.org/10.1371/journal.pcbi.1006790 PMCID: PMC6380598
</li><li><b>Mungall CJ</b>, Koehler S, Robinson P, Holmes I, Haendel M. k-BOOM: A Bayesian approach to ontology structure inference, with applications in disease ontology construction. bioRxiv. 2019. https://www.biorxiv.org/content/10.1101/048843v3
</li><li>Giglio M, Tauber R, Nadendla S, Munro J, Olley D, Ball S, Mitraka E, Schriml LM, Gaudet P, Hobbs ET, Erill I, Siegele DA, Hu JC, <b>Mungall C</b>, Chibucos MC. ECO, the Evidence & Conclusion Ontology: community standard for evidence information. Nucleic Acids Res. 2019 Jan 8;47(D1):D1186–D1194. http://dx.doi.org/10.1093/nar/gky1036 PMCID: PMC6323956
</li><li>Haendel MA, McMurry JA, Relevo R, <b>Mungall CJ</b>, Robinson PN, Chute CG. A Census of Disease Ontologies. Annu Rev Biomed Data Sci. Annual Reviews; 2018 Jul 20;1(1):305–331. https://doi.org/10.1146/annurev-biodatasci-080917-013459
</li><li>Klopfenstein DV, Zhang L, Pedersen BS, Ramírez F, Warwick Vesztrocy A, Naldi A, <b>Mungall CJ</b>, Yunes JM, Botvinnik O, Weigel M, Dampier W, Dessimoz C, Flick P, Tang H. GOATOOLS: A Python library for Gene Ontology analyses. Sci Rep. 2018 Jul 18;8(1):10872. http://dx.doi.org/10.1038/s41598-018-28948-z PMCID: PMC6052049
</li><li>Vasilevsky NA, Foster ED, Engelstad ME, Carmody L, Might M, Chambers C, Dawkins HJS, Lewis J, Della Rocca MG, Snyder M, Boerkoel CF, Rath A, Terry SF, Kent A, Searle B, Baynam G, Jones E, Gavin P, Bamshad M, Chong J, Groza T, Adams D, Resnick AC, Heath AP, <b>Mungall C</b>, Holm IA, Rageth K, Brownstein CA, Shefchek K, McMurry JA, Robinson PN, Köhler S, Haendel MA. Plain-language medical vocabulary for precision diagnosis. Nat Genet. 2018 Apr;50(4):474–476. http://dx.doi.org/10.1038/s41588-018-0096-x PMCID: PMC6258202
</li><li>Tang H, <b>Mungall CJ</b>, Mi H, Thomas PD. GOTaxon: Representing the evolution of biological functions in the Gene Ontology. arXiv [q-bio.PE]. 2018. http://arxiv.org/abs/1802.06004
</li><li>Matentzoglu N, Malone J, <b>Mungall C</b>, Stevens R. MIRO: guidelines for minimum information for the reporting of an ontology. J Biomed Semantics. 2018 Jan 18;9(1):6. http://dx.doi.org/10.1186/s13326-017-0172-7 PMCID: PMC5774126
</li><li>Cooper L, Meier A, Laporte M-A, Elser JL, <b>Mungall C</b>, Sinn BT, Cavaliere D, <b>Carbon S</b>, <b>Dunn N</b>A, Smith B, Qu B, Preece J, Zhang E, Todorovic S, Gkoutos G, Doonan JH, Stevenson DW, Arnaud E, Jaiswal P. The Planteome database: an integrated resource for reference ontologies, plant genomics and phenomics. Nucleic Acids Res. 2018 Jan 4;46(D1):D1168–D1180. http://dx.doi.org/10.1093/nar/gkx1152 PMCID: PMC5753347
</li><li>Haendel MA, Mcmurry JA, Relevo R, <b>Mungall CJ</b>, Robinson PN, Chute CG. Annual Review of Biomedical Data Science A Census of Disease Ontologies. 2018;(April):305–331.
</li><li>GO Consortium. Gene Ontology Resource: 20 years and still GOing strong. Nucleic Acids Res. Oxford University Press; 2018;47(D1):D330–D338.
</li><li>Köhler S, Carmody L, Vasilevsky N, Jacobsen JOB, Danis D, Gourdine J-P, Gargano M, <b>Harris NL</b>, Matentzoglu N, McMurry JA, Others. Expansion of the Human Phenotype Ontology (HPO) knowledge base and resources. Nucleic Acids Res. Oxford University Press; 2018;47(D1):D1018–D1027.
</li><li>Balhoff JP, Good BM, <b>Carbon S</b>, <b>Mungall CJ</b>. Arachne: an OWL RL reasoner applied to Gene Ontology Causal Activity Models (and beyond). Zenodo; 2018. http://dx.doi.org/10.5281/ZENODO.2397192
</li><li><b>Mungall CJ</b>, McMurry JA, Köhler S, Balhoff JP, Borromeo C, Brush M, <b>Carbon S,</b> Conlin T, Dunn N, Engelstad M, Foster E, Gourdine JP, Jacobsen JOB, Keith D, Laraway B, Lewis SE, NguyenXuan J, Shefchek K, Vasilevsky N, Yuan Z, Washington N, Hochheiser H, Groza T, Smedley D, Robinson PN, Haendel MA. The Monarch Initiative: an integrative data and analytic platform connecting phenotypes to genotypes across species. Nucleic Acids Res. 2017 Jan 4;45(D1):D712–D722. http://dx.doi.org/10.1093/nar/gkw1128 PMCID: PMC5210586
</li><li>Holmes IH, <b>Mungall CJ</b>. BioMake: a GNU make-compatible utility for declarative workflow management. Bioinformatics. 2017 Nov 1;33(21):3502–3504. http://dx.doi.org/10.1093/bioinformatics/btx306 PMCID: PMC5860158
</li><li>Osumi-Sutherland D, Courtot M, Balhoff JP, <b>Mungall C</b>. Dead simple OWL design patterns. J Biomed Semantics. 2017 Jun 5;8(1):18. http://dx.doi.org/10.1186/s13326-017-0126-0 PMCID: PMC5460348
</li><li><b>Mungall CJ</b>, Holmes IH. WTFgenes: What’s The Function of these genes? Static sites for model-based gene set analysis. F1000Res. 2017 Apr 4. https://f1000research.com/articles/6-423/v1/pdf
</li><li>Köhler S, Vasilevsky NA, Engelstad M, Foster E, McMurry J, Aymé S, Baynam G, Bello SM, Boerkoel CF, Boycott KM, Brudno M, Buske OJ, Chinnery PF, Cipriani V, Connell LE, Dawkins HJS, DeMare LE, Devereau AD, de Vries BBA, Firth HV, Freson K, Greene D, Hamosh A, Helbig I, Hum C, Jähn JA, James R, Krause R, F Laulederkind SJ, Lochmüller H, Lyon GJ, Ogishima S, Olry A, Ouwehand WH, Pontikos N, Rath A, Schaefer F, Scott RH, Segal M, Sergouniotis PI, Sever R, Smith CL, Straub V, Thompson R, Turner C, Turro E, Veltman MWM, Vulliamy T, Yu J, von Ziegenweidt J, Zankl A, Züchner S, Zemojtel T, Jacobsen JOB, Groza T, Smedley D, <b>Mungall CJ</b>, Haendel M, Robinson PN. The Human Phenotype Ontology in 2017. Nucleic Acids Res. 2017 Jan 4;45(D1):D865–D876. http://dx.doi.org/10.1093/nar/gkw1039 PMCID: PMC5210535
</li><li>Munoz-Torres M, <b>Carbon S</b>. Get GO! Retrieving GO Data Using AmiGO, QuickGO, API, Files, and Tools. Methods Mol Biol. 2017;1446:149–160. http://dx.doi.org/10.1007/978-1-4939-3743-1_11 PMID: 27812941
</li><li>Diehl AD, Meehan TF, Bradford YM, Brush MH, Dahdul WM, Dougall DS, He Y, Osumi-Sutherland D, Ruttenberg A, Sarntivijai S, Van Slyke CE, Vasilevsky NA, Haendel MA, Blake JA, <b>Mungall CJ</b>. The Cell Ontology 2016: enhanced content, modularization, and ontology interoperability. J Biomed Semantics. 2016 Jul 4;7(1):44. http://dx.doi.org/10.1186/s13326-016-0088-7 PMCID: PMC4932724
</li><li>Buttigieg PL, Pafilis E, Lewis SE, Schildhauer MP, Walls RL, <b>Mungall CJ</b>. The environment ontology in 2016: bridging domains with increased scope, semantic density, and interoperation. J Biomed Semantics. 2016 Sep 23;7(1):57. http://dx.doi.org/10.1186/s13326-016-0097-6 PMCID: PMC5035502

</li><li>Buske OJ, Schiettecatte F, Hutton B, Dumitriu S, Misyura A, Huang L, Hartley T, Girdea M, Sobreira N, <b>Mungall C</b>, Brudno M. The Matchmaker Exchange API: automating patient matching through the exchange of structured phenotypic and genotypic profiles. Hum Mutat. 2015 Oct;36(10):922–927. http://dx.doi.org/10.1002/humu.22850 PMCID: PMC4775166

</li><li><b>Mungall CJ</b>, Washington NL, Nguyen-Xuan J, Condit C, Smedley D, Köhler S, Groza T, Shefchek K, Hochheiser H, Robinson PN, Lewis SE, Haendel MA. Use of model organism and disease databases to support matchmaking for human disease gene discovery. Hum Mutat. 2015 Oct;36(10):979–984. http://dx.doi.org/10.1002/humu.22857 PMCID: PMC5473253

</li><li>Dietze H, Berardini TZ, Foulger RE, Hill DP, Lomax J, Osumi-Sutherland D, Roncaglia P, <b>Mungall CJ</b>. TermGenie -- a web-application for pattern-based ontology class generation. J Biomed Semantics. 2014 Dec 11;5(1):48. https://doi.org/10.1186/2041-1480-5-48

</li><li>Haendel MA, Balhoff JP, Bastian FB, Blackburn DC, Blake JA, Bradford Y, Comte A, Dahdul WM, Dececchi TA, Druzinsky RE, Hayamizu TF, Ibrahim N, Lewis SE, Mabee PM, Niknejad A, Robinson-Rechavi M, Sereno PC, <b>Mungall CJ</b>. Unification of multi-species vertebrate anatomy ontologies for comparative biology in Uberon. J Biomed Semantics. 2014 May 19;5:21. http://dx.doi.org/10.1186/2041-1480-5-21 PMCID: PMC4089931
</li><li>Köhler S, Doelken SC, <b>Mungall CJ</b>, Bauer S, Firth HV, Bailleul-Forestier I, Black GCM, Brown DL, Brudno M, Campbell J, FitzPatrick DR, Eppig JT, Jackson AP, Freson K, Girdea M, Helbig I, Hurst JA, Jähn J, Jackson LG, Kelly AM, Ledbetter DH, Mansour S, Martin CL, Moss C, Mumford A, Ouwehand WH, Park S-M, Riggs ER, Scott RH, Sisodiya S, Van Vooren S, Wapner RJ, Wilkie AOM, Wright CF, Vulto-van Silfhout AT, de Leeuw N, de Vries BBA, Washingthon NL, Smith CL, Westerfield M, Schofield P, Ruef BJ, Gkoutos GV, Haendel M, Smedley D, Lewis SE, Robinson PN. The Human Phenotype Ontology project: linking molecular biology and disease through phenotype data. Nucleic Acids Res. 2014 Jan;42(Database issue):D966–74. http://dx.doi.org/10.1093/nar/gkt1026 PMCID: PMC3965098
</li><li>Lee E, Helt GA, <b>Reese JT</b>, Munoz-Torres MC, Childers CP, Buels RM, Stein L, Holmes IH, Elsik CG, Lewis SE (2013). Web Apollo: a web-based genomic annotation editing platform. Genome Biol, 2013 Aug 30;14(8):R93. http://dx.doi.org/10.1186/gb-2013-14-8-r93 PMCID: PMC4053811

</li><li><b>Mungall CJ</b>, Torniai C, Gkoutos GV, Lewis SE, Haendel MA. Uberon, an integrative multi-species anatomy ontology. Genome Biol. 2012 Jan 31;13(1):R5. http://dx.doi.org/10.1186/gb-2012-13-1-r5 PMCID: PMC3334586

</li><li><b>Mungall CJ</b>, Gkoutos GV, Smith CL, Haendel MA, Lewis SE, Ashburner M. Integrating phenotype ontologies across multiple species. Genome Biol. 2010 Jan 8;11(1):R2. http://dx.doi.org/10.1186/gb-2010-11-1-r2 PMCID: PMC2847714
</li><li>Washington NL, Haendel MA, <b>Mungall CJ</b>, Ashburner M, Westerfield M, Lewis SE. Linking human diseases to animal models using ontology-based phenotype annotation. PLoS Biol. 2009 Nov;7(11):e1000247. http://dx.doi.org/10.1371/journal.pbio.1000247 PMCID: PMC2774506
</li><li>Skinner ME, Uzilov AV, Stein LD, <b>Mungall CJ</b>, Holmes IH. JBrowse: a next-generation genome browser. Genome Res. 2009 Sep;19(9):1630–1638. http://dx.doi.org/10.1101/gr.094607.109 PMCID: PMC2752129
</li><li>Haendel MA, Gkoutos GG, Lewis SE, <b>Mungall C</b>. Uberon: towards a comprehensive multi-species anatomy ontology. 2009. http://precedings.nature.com/documents/3592/version/1
</li><li>Lee E, <b>Harris N</b>, Gibson M, Chetty R, Lewis S. Apollo: a community resource for genome annotation editing. Bioinformatics. Oxford University Press; 2009;25(14):1836–1837.
</li><li><b>Carbon S</b>, Ireland A, <b>Mungall CJ</b>, Shu S, Marshall B, Lewis S, AmiGO Hub, Web Presence Working Group. AmiGO: online access to ontology and annotation data. Bioinformatics. 2009 Jan 15;25(2):288–289. http://dx.doi.org/10.1093/bioinformatics/btn615 PMCID: PMC2639003
</li><li>Smith B, Ashburner M, Rosse C, Bard J, Bug W, Ceusters W, Goldberg LJ, Eilbeck K, Ireland A, <b>Mungall CJ</b>, OBI Consortium, Leontis N, Rocca-Serra P, Ruttenberg A, Sansone S-A, Scheuermann RH, Shah N, Whetzel PL, Lewis S. The OBO Foundry: coordinated evolution of ontologies to support biomedical data integration. Nat Biotechnol. 2007 Nov;25(11):1251–1255. http://dx.doi.org/10.1038/nbt1346 PMCID: PMC2814061
</li><li>Eilbeck K, Lewis SE, <b>Mungall CJ</b>, Yandell M, Stein L, Durbin R, Ashburner M. The Sequence Ontology: a tool for the unification of genome annotations. Genome Biol. 2005 Apr 29;6(5):R44. http://dx.doi.org/10.1186/gb-2005-6-5-r44 PMCID: PMC1175956
</li><li>Harris MA, Clark J, Ireland A, Lomax J, Ashburner M, Foulger R, Eilbeck K, Lewis S, Marshall B, <b>Mungall C</b>, Richter J, Rubin GM, Blake JA, Bult C, Dolan M, Drabkin H, Eppig JT, Hill DP, Ni L, Ringwald M, Balakrishnan R, Cherry JM, Christie KR, Costanzo MC, Dwight SS, Engel S, Fisk DG, Hirschman JE, Hong EL, Nash RS, Sethuraman A, Theesfeld CL, Botstein D, Dolinski K, Feierbach B, Berardini T, Mundodi S, Rhee SY, Apweiler R, Barrell D, Camon E, Dimmer E, Lee V, Chisholm R, Gaudet P, Kibbe W, Kishore R, Schwarz EM, Sternberg P, Gwinn M, Hannick L, Wortman J, Berriman M, Wood V, de la Cruz N, Tonellato P, Jaiswal P, Seigfried T, White R, Gene Ontology Consortium. The Gene Ontology (GO) database and informatics resource. Nucleic Acids Res. 2004 Jan 1;32(Database issue):D258–61. http://dx.doi.org/10.1093/nar/gkh036 PMCID: PMC308770
</li><li>FlyBase Consortium. The FlyBase database of the Drosophila genome projects and community literature. Nucleic Acids Res. 2003 Jan 1;31(1):172–175. https://www.ncbi.nlm.nih.gov/pubmed/12519974 PMCID: PMC165541
</li><li>Stein LD, <b>Mungall C</b>, Shu S, Caudy M, Mangone M, Day A, Nickerson E, Stajich JE, Harris TW, Arva A, Lewis S. The generic genome browser: a building block for a model organism system database. Genome Res. 2002 Oct;12(10):1599–1610. http://dx.doi.org/10.1101/gr.403602 PMCID: PMC187535
</li><li><b>Mungall CJ</b>, Misra S, Berman BP, Carlson J, Frise E, <b>Harris N</b>, Marshall B, Shu S, Kaminker JS, Prochnik SE, Others. An integrated computational pipeline and database to support whole-genome sequence annotation. Genome Biol. BioMed Central; 2002;3(12):research0081–1.
</li><li>Gene Ontology Consortium. Creating the gene ontology resource: design and implementation. Genome Res. 2001 Aug;11(8):1425–1433. http://dx.doi.org/10.1101/gr.180801 PMCID: PMC311077
</li><li>Lewis SE, Searle SMJ, <b>Harris N</b>, Gibson M, Iyer V, Richter J, Wiel C, Bayraktaroglu L, Birney E, Crosby MA, Others (2002). Apollo: a sequence annotation editor. Genome Biol. BioMed Central; 2002;3(12):research0082–1.
</li><li>Reese MG, Hartzell G, <b>Harris NL</b>, Ohler U, Abril JF, Lewis SE. Genome annotation assessment in Drosophila melanogaster. Genome Res. Cold Spring Harbor Lab; 2000;10(4):483–501.
</li><li>Rubin GM, Yandell MD, Wortman JR, Gabor Miklos GL, Nelson CR, Hariharan IK, Fortini ME, Li PW, Apweiler R, Fleischmann W, Cherry JM, Henikoff S, Skupski MP, Misra S, Ashburner M, Birney E, Boguski MS, Brody T, Brokstein P, Celniker SE, Chervitz SA, Coates D, Cravchik A, Gabrielian A, Galle RF, Gelbart WM, George RA, Goldstein LS, Gong F, Guan P, <b>Harris NL</b>, Hay BA, Hoskins RA, Li J, Li Z, Hynes RO, Jones SJ, Kuehl PM, Lemaitre B, Littleton JT, Morrison DK, <b>Mungall C</b>, O’Farrell PH, Pickeral OK, Shue C, Vosshall LB, Zhang J, Zhao Q, Zheng XH, Lewis S. Comparative genomics of the eukaryotes. Science. 2000 Mar 24;287(5461):2204–2215. https://www.ncbi.nlm.nih.gov/pubmed/10731134 PMCID: PMC2754258
</li><li>Adams MD, Celniker SE, Holt RA, Evans CA, Gocayne JD, Amanatides PG, Scherer SE, Li PW, Hoskins RA, Galle RF, George RA, Lewis SE, Richards S, Ashburner M, Henderson SN, Sutton GG, Wortman JR, Yandell MD, Zhang Q, Chen LX, Brandon RC, Rogers YH, Blazej RG, Champe M, Pfeiffer BD, Wan KH, Doyle C, Baxter EG, Helt G, Nelson CR, Gabor GL, Abril JF, Agbayani A, An HJ, Andrews-Pfannkoch C, Baldwin D, Ballew RM, Basu A, Baxendale J, Bayraktaroglu L, Beasley EM, Beeson KY, Benos PV, Berman BP, Bhandari D, Bolshakov S, Borkova D, Botchan MR, Bouck J, Brokstein P, Brottier P, Burtis KC, Busam DA, Butler H, Cadieu E, Center A, Chandra I, Cherry JM, Cawley S, Dahlke C, Davenport LB, Davies P, de Pablos B, Delcher A, Deng Z, Mays AD, Dew I, Dietz SM, Dodson K, Doup LE, Downes M, Dugan-Rocha S, Dunkov BC, Dunn P, Durbin KJ, Evangelista CC, Ferraz C, Ferriera S, Fleischmann W, Fosler C, Gabrielian AE, Garg NS, Gelbart WM, Glasser K, Glodek A, Gong F, Gorrell JH, Gu Z, Guan P, Harris M, <b>Harris NL</b>, Harvey D, Heiman TJ, Hernandez JR, Houck J, Hostin D, Houston KA, Howland TJ, Wei MH, Ibegwam C, Jalali M, Kalush F, Karpen GH, Ke Z, Kennison JA, Ketchum KA, Kimmel BE, Kodira CD, Kraft C, Kravitz S, Kulp D, Lai Z, Lasko P, Lei Y, Levitsky AA, Li J, Li Z, Liang Y, Lin X, Liu X, Mattei B, McIntosh TC, McLeod MP, McPherson D, Merkulov G, Milshina NV, Mobarry C, Morris J, Moshrefi A, Mount SM, Moy M, Murphy B, Murphy L, Muzny DM, Nelson DL, Nelson DR, Nelson KA, Nixon K, Nusskern DR, Pacleb JM, Palazzolo M, Pittman GS, Pan S, Pollard J, Puri V, Reese MG, Reinert K, Remington K, Saunders RD, Scheeler F, Shen H, Shue BC, Sidén-Kiamos I, Simpson M, Skupski MP, Smith T, Spier E, Spradling AC, Stapleton M, Strong R, Sun E, Svirskas R, Tector C, Turner R, Venter E, Wang AH, Wang X, Wang ZY, Wassarman DA, Weinstock GM, Weissenbach J, Williams SM, WoodageT, Worley KC, Wu D, Yang S, Yao QA, Ye J, Yeh RF, Zaveri JS, Zhan M, Zhang G, Zhao Q, Zheng L, Zheng XH, Zhong FN, Zhong W, Zhou X, Zhu S, Zhu X, Smith HO, Gibbs RA, Myers EW, Rubin GM, Venter JC. The genome sequence of Drosophila melanogaster. Science. 2000 Mar 24;287(5461):2185–2195. https://www.ncbi.nlm.nih.gov/pubmed/10731132 PMID: 10731132

</li>
</ol>
