---
layout: project
title: Gene Ontology
active: yes
funding: NIH HG002273
homepage: http://geneontology.org
twitter: news4go
---

The [Gene Ontology](http://geneontology.org) (GO) project is a collaborative effort to address the need for consistent descriptions of gene products in different databases. The GO collaborators are developing three structured, controlled vocabularies (ontologies) that describe gene products in terms of their associated biological processes, cellular components and molecular functions in a species-independent manner. There are three separate aspects to this effort: first, we write and maintain the ontologies themselves; second, we make cross-links between the ontologies and the genes and gene products in the collaborating databases, and third, we develop tools that facilitate the creation, maintainance and use of ontologies.

The use of GO terms by several collaborating databases facilitates uniform queries across them. The controlled vocabularies are structured so that you can query them at different levels: for example, you can use GO to find all the gene products in the mouse genome that are involved in signal transduction, or you can zoom in on all the receptor tyrosine kinases. This structure also allows annotators to assign properties to gene products at different levels, depending on how much is known about a gene product.

## Our Role

BBOP has been a member of the GO consortium since its inception in 1999, for which Suzanna Lewis was a founding member (with Michael Ashburner, Judy Blake, and Mike Cherry). BBOP currently contributes to the [GO consortium](http://www.geneontology.org/GO.consortiumlist.shtml) in several ways: project vision, ontology development, and in the development of software tools to view and edit GO ontologies ([OBO-Edit](http://www.oboedit.org/)) and their annotations ([AmiGO](http://amigo.geneontology.org/cgi-bin/amigo/go.cgi)).

## References

 * Mungall, C. J., Dietze, H., & Osumi-Sutherland, D. (2014). Use of OWL within the Gene Ontology. In M. Keet & V. Tamma (Eds.), Proceedings of the 11th International Workshop on OWL: Experiences and Directions (OWLED 2014) (pp. 25–36). Riva del Garda, Italy, October 17-18, 2014. http://doi.org/10.1101/010090
 * Thomas, P. D., Wood, V., Mungall, C. J., Lewis, S. E., & Blake, J. A. (2012). On the Use of Gene Ontology Annotations to Assess Functional Similarity among Orthologs and Paralogs: A Short Report. PLoS Computational Biology, 8(2), e1002386. http://doi.org/10.1371/journal.pcbi.1002386
 * Hill, D. P., Adams, N., Bada, M., Batchelor, C., Berardini, T. Z., Dietze, H., … Lomax, J. (2013). Dovetailing biology and chemistry: integrating the Gene Ontology with the ChEBI chemical ontology. BMC Genomics, 14(1), 513. http://doi.org/10.1186/1471-2164-14-513
 * Deegan, J., Dimmer, E., & Mungall, C. J. (2010). Formalization of taxon-based constraints to detect inconsistencies in annotation and ontology development. BMC Bioinformatics, 11(1), 530. http://doi.org/10.1186/1471-2105-11-530
 * Huntley, R. P., Harris, M. a, Alam-Faruque, Y., Blake, J. a, Carbon, S., Dietze, H., … Mungall, C. J. (2014). A method for increasing expressivity of Gene Ontology annotations using a compositional approach. BMC Bioinformatics, 15(1), 155. http://doi.org/10.1186/1471-2105-15-155
 * Mungall, C. J., Bada, M., Berardini, T. Z., Deegan, J., Ireland, A., Harris, M. A., … Lomax, J. (2011). Cross-product extensions of the Gene Ontology. Journal of Biomedical Informatics, 44(1), 80–86. http://doi.org/10.1016/j.jbi.2010.02.002
 
