---
layout: project
title: Exomiser
active: yes
homepage: https://exomiser.monarchinitiative.org/exomiser/
funding: 1R01HD103805-01
github: https://github.com/exomiser/Exomiser
logo: https://exomiser.monarchinitiative.org/exomiser/logo.png
travis: 
docs: 
twitter: 
---

# Increasing the Yield and Utility of Pediatric Genomic Medicine with Exomiser

Exomiser is a tool that finds potential disease-causing variants from whole-exome or whole-genome sequencing data. 
Starting from a VCF file and a set of phenotypes encoded using the Human Phenotype Ontology (HPO) it annotates, filters and prioritises likely causative variants.
It does this based on user-defined criteria such as a variant's predicted pathogenicity, frequency of occurrence in a population and
also how closely the given phenotype matches the known phenotype of diseased genes from human and model organism data.

In this collaboration with the Jackson Laboratory, we are extending Exomiser to work better in pediatric genomic medicine.
Rare genetic diseases are collectively a common cause of morbidity in children.
Precision clinical management is often hindered by the difficulties in making an accurate and precise diagnosis, even with new DNA sequencing technologies
that can scan all genes or even all genetic material (the ‘genome’) at once.
We are extending xomiser to meet current challenges and opportunities in pediatrics by adding algorithms for new genomic technologies, analyzing molecular pathways involved in intellectual disability,
and providing approaches to analyze groups of unsolved cases to identify novel disease genes.
